### PR TITLE
Migrate 'close' command to ArgParse

### PR DESCRIPTION
**Pull request by t-wissmann**
The usual migration commit: use ArgParse for argument parsing and
command line completions, add tests for it, remove the old
implementation and completion code.

### CODE REVIEW

File: tests/test_clients.py [hunk 1]
```diff
@@ -49,11 +49,50 @@ def test_close_without_clients(hlwm):
 
 
 @pytest.mark.parametrize("running_clients_num", [1, 2, 3, 4])
-def test_close(hlwm, running_clients_num):
+def test_close_for_multiple_clients(hlwm, running_clients_num):
```
Comment (blarz) [inline tests/test_clients.py:52]: This test and some others related to the "close" command do not assert anything.
Is this inteded? How is it tested, that the clients are actually closed?

**Review by blarz — COMMENTED** [review]
Looks good, just one question about the tests in general

File: tests/test_clients.py (hunk 1 above)
Comment (t-wissmann) [reply tests/test_clients.py:52]: I also can't remember what's the purpose of this test. It's inconvenient to wait for a client to close if we don't have it's process handle. Maybe this test case just checked that hlwm does not crash. Maybe I should just extend `test_close_focused_client` by other unfocused clients to verify that 'close' only closes the focused one.

Comment (t-wissmann) [reply tests/test_clients.py]: I've removed this incomplete test and extended the other test where it is actually verified that the focused client closes.

Comment (blarz) [reply tests/test_clients.py]: Thanks for updating the test!
Should we add checks for the client to be closed in the unfocused and unmanaged case as well?

Comment (t-wissmann) [reply tests/test_clients.py]: In the unfocused and unmanaged case, it is already checked that it is closed (if something is closed, then there is `proc.wait(10)` in the test.)

Comment (blarz) [reply tests/test_clients.py]: Aaah, yes, you're right. I was looking for asserts, but the proc.wait() is the assert in that case

**Review by blarz — APPROVED** [review]

**Pull request merged**